katello/katello-change-hostname

```diff
@@ -0,0 +1,190 @@
+require 'optparse'
+
+DEFAULT_PROGRAM = "foreman"
+DEFAULT_SCENARIO = "katello"
+DEFAULT_SSL_DIR = "/root/ssl-build"
+
+def is_rhel_version?(version)
+  `cat /etc/redhat-release`[/(\d+\.)/].include?(version)
+end
+
+def yesno
+  begin
+    system("stty raw -echo")
+    str = STDIN.getc
+  ensure
+    system("stty -raw echo")
+  end
+  if str.chr.downcase == "y"
+    return true
+  elsif str.chr.downcase == "n"
+    return false
+  else
+    puts "Invalid Character. Try again: [y/n]"
+    yesno
+  end
+end
+
+def timestamp
+  Time.now.strftime("%Y%m%d%H%M")
+end
+
+def get_hostname
+  if is_rhel_version?("6")
+    hostname = `hostname`
+  else
+    hostname = Socket.gethostname
+  end
+  hostname.chomp 
+end
+
+options = {}
+options[:program] = DEFAULT_PROGRAM
+options[:scenario] = DEFAULT_SCENARIO
+options[:ssl_dir] = DEFAULT_SSL_DIR
+
+opt_parser = OptionParser.new do |opt|
+  opt.banner = "Usage: katello-change-hostname HOSTNAME [OPTIONS]"
+  opt.separator  ""
+  opt.separator  "Example:"
+  opt.separator  " katello-change-hostname foo.example.com"
+  opt.separator  ""
+  opt.separator  "Options"
+
+  opt.on("-s","--program PROGRAM","name of the scenario you are modifying (defaults to #{DEFAULT_PROGRAM})") do |program|
+    options[:program] = program
+  end
+
+  opt.on("-s","--scenario SCENARIO","name of the scenario you are modifying (defaults to #{DEFAULT_SCENARIO})") do |scenario|
+    options[:scenario] = scenario
+  end
+
+  opt.on("-d","--disable-system-checks","runs the installer with --disable-system-checks") do |system_check|
+    options[:system_check] = true
+  end
+
+  opt.on("-l","--ssl-dir SSL_DIR","specify the ssl cert directory to replace (defaults to #{DEFAULT_SSL_DIR})") do |ssl_dir|
+    options[:ssl_dir] = ssl_dir
+  end
+
+ opt.on("-h","--help","help") do
+    puts opt_parser
+    exit
+  end
+end
+opt_parser.parse!
+
+raise 'Must run as root' unless Process.uid == 0
+
+if options[:scenario] == "katello"
+  proxy = "Smart Proxy"
+else
+  proxy = "Capsule"
+end
+
+if ARGV[0] && ARGV.count >= 1
+  new_hostname = ARGV[0]
+else
+  puts opt_parser
+  exit
+end
+
+STDOUT.puts("\nA hostname change requires a manual change in the program. " \
+	     "Please log into UI and rename the default #{proxy} (under Infrastructure -> #{proxy}) " \
+             "to #{new_hostname} using both the name and url fields. " \
+             "After this script completes, you will have to re-register any capsules and #{options[:scenario]} clients. " \
+             "All services will be stopped and started during this process. This script will modify your system!")
+STDOUT.print("\nHave you made this necessary change and want to proceed with changing your hostname to #{new_hostname}? [y/n]")
+response = yesno
+STDOUT.puts ""
+
+unless response
+  STDOUT.puts "Hostname change aborted, no changes have been made to your system"
+  exit
+end
+
+# Get the hostname from your system
+old_hostname = get_hostname
+
+if is_rhel_version?("6")
```

Comment:
I suppose if we are backporting this, this makes sense, however we don't support RHEL 6 on 3.3+.